### ISSUE TITLE
Cannot install from local git repo 

### ISSUE THREAD

**Issue by yggdr** [pypa/pipx#1583]
Trying a few variations of `pipx install <SPEC>` always yields `Unable to parse package spec:`
I tried
`/path/to/repo#<HASH>`, `git:///path/to/repo#<HASH>`, `git+file:///path/to/repo#<HASH>`, `git+files:///path/to/repo#<HASH>`, and the same with `.git` added to end of the path.
Are local git operations unsupported?

**Comment by huxuan** [comment]
Instead of installing from local git repo, I would suggest to install from local directory directly. Just `pipx install /path/to/repo` should work. You can even use `pipx install -e /path/to/repo` and in this way, pipx will automatically change the installed application according to the commit of the local directory you checked out. I often use this to debug pipx. :-)

Will this resolve your problem?

**Comment by yggdr** [comment]
No, I specifically wanted to install from a certain commit in a repository, not from a directory. The repo happens to live in a local directory.

**Comment by huxuan** [comment]
AFAIK, there is no standarized URI format with commit hash for local git repos, so I do not think pipx should (and easy for pipx to) "create" it.

For your use cases, I would suggest to `git checkout` the certain commit you want locally and `pipx install` or `pipx install -e` just as I have mentioned.

I will close this issue and feel free to reopen it if necessary.

**Comment by yggdr** [comment]
It's part of PyPA: https://pip.pypa.io/en/latest/topics/vcs-support/

**Comment by huxuan** [comment]
Sorry, I will further look into it. Thanks for the reference. Reopen the issue.

**Comment by dukecat0** [comment]
It looks like we need to update the logic here:
https://github.com/pypa/pipx/blob/36d632d8597b397b3d800b8d6ad78cc9a6fb4678/src/pipx/package_specifier.py#L51

We should support path to local git repo as well.
